### ISSUE TITLE
PX4 in SITL

### ISSUE THREAD

**Issue by johan606303** [microsoft/AirSim#2647]
Hi.
I am trying to set PX4 in SITL mode.
The commands : `git checkout v1.10.1` and `git checkout v1.10.2` .
So I tried with `git checkout v1.11.0-beta1` and I arrived to : 
![px4_1](https://user-images.githubusercontent.com/62846458/80799001-fb49bd80-8ba5-11ea-96a9-8bfd799fd95c.PNG)

So after I modified `Settings.json` : 

![px4_2](https://user-images.githubusercontent.com/62846458/80799056-23392100-8ba6-11ea-8674-05a3b78bc486.PNG)

But when I run a project, `run-console.bat` returns the following issues : 
![px4_5](https://user-images.githubusercontent.com/62846458/80799825-2d5c1f00-8ba8-11ea-8f47-2735452fa45d.PNG)
 And on UE4 I have the following messages : 
![px4_3](https://user-images.githubusercontent.com/62846458/80799874-4d8bde00-8ba8-11ea-8f12-f69c7c2f9645.PNG)
What do I have to do else to control the drone using Python ? 

Thanks for your help.

**Comment by rajat2004** [comment]
Are you running the latest AirSim master? If not, then try with that, several issues with PX4 have been fixed very recently

**Comment by johan606303** [comment]
Do you mean run the command  `git rebase origin/master` ?

**Comment by rajat2004** [comment]
git fetch origin
git rebase origin/master

**Comment by johan606303** [comment]
Well I ran those commands, it returned to me :
![px4_6](https://user-images.githubusercontent.com/62846458/80823802-8a25fc80-8bdd-11ea-87c1-8c2965bd6d1c.PNG)

So the issue is a bit different. But I forgot to mention that when I run a project, when UE4 appear and I press `Play`, VS2019 says that i meant `breakpoints`
![px4_7](https://user-images.githubusercontent.com/62846458/80823988-dcffb400-8bdd-11ea-9952-ac4288c52a80.PNG)
![px4_8](https://user-images.githubusercontent.com/62846458/80823994-dffaa480-8bdd-11ea-8dfa-4001c42a4da4.PNG)

**Comment by rajat2004** [comment]
You might be running into the problem discussed in https://github.com/microsoft/AirSim/issues/2583
@lovettchris would have a better idea
Just to confirm, you're not running in HITL and only SITL correct? Also, are you using any joystick?

**Comment by johan606303** [comment]
It's for SITL, and I don't use joystics, I don't have any remote controle. I want to controle just through executing codes, at least for now.

**Comment by rajat2004** [comment]
Just to check if the current latest master is being used, could you put the last commit id
`git log` should give the list of commits

Also, did you run `build.cmd` and then `update_from_git.bat` to update the plugin

**Comment by lovettchris** [comment]
The heap corruption in AirSim was fixed with this PR #2634, can you make sure you have those bits installed ?  I built this fix into these binaries, and I've tested those binaries with PX4 v1.11.0-beta1:

- [AirSim Neighborhood binary (1.3gb)](https://drive.google.com/file/d/1DIleir72Jzfx3oU2i8fEHNBI-37UfXpA/view?usp=sharing) 
- [AirSim Blocks binary (153mb)](https://drive.google.com/file/d/1OMvLO2LcNI5KxYovYKTWj9XGGXHYXvNC/view?usp=sharing) 

I also see in your screen shot where it talks about lighting needs to be rebuilt, so I wonder if Unreal is still compiling shaders in the background, which could create some timing problems...  My pre-compiled binaries will not have that problem.  This shows a python script flying the above AirSim Neighborhood using PX4 SITL v1.11.0-beta1:

![image](https://user-images.githubusercontent.com/18707114/80833047-a48eb480-8ba2-11ea-8bdd-1b6c7b8d6b35.png)

**Comment by johan606303** [comment]
When I run `git log` I have : 

![px4_9](https://user-images.githubusercontent.com/62846458/80857847-8d5ccf00-8c55-11ea-9e2f-db1de998c54b.PNG)

**Comment by johan606303** [comment]
Well I ran commands `build.cmd`, and `update_from_git.bat`. Then when I ran the project Blocks, on PX4 `run-console.bat` I have : 
![px4_13](https://user-images.githubusercontent.com/62846458/80858333-0c9fd200-8c59-11ea-834c-46bcce1be0f4.PNG)

and on UE4 : 

![px4_12](https://user-images.githubusercontent.com/62846458/80858370-3658f900-8c59-11ea-99c7-106167344ac8.PNG)
 Then if I try to run api `armDisarm(True)`
![px4_15](https://user-images.githubusercontent.com/62846458/80858407-6bfde200-8c59-11ea-81d8-047ba082da9e.PNG)

and on UE4 I have : 

![px4_14](https://user-images.githubusercontent.com/62846458/80858418-791ad100-8c59-11ea-9b43-b06c1fc397e0.PNG)

**Comment by bys1123** [comment]
@lovettchris Thanks, I'll test these binary.

**Comment by johan606303** [comment]
@bys1123 Hi, does it work for you ?

**Comment by bys1123** [comment]
https://github.com/microsoft/AirSim/pull/2649
@johan606303 Not yet, here is new PR and new binaries, might works.

**Comment by johan606303** [comment]
Ok, thank you for the link.

**Comment by johan606303** [comment]
@lovettchris Ok. I tried to run with Blocks or Neighborhoood, that @bys1123 sent me. But I still have those issues, and I con't understand why. On `run-console.bat` : 

![px4_22](https://user-images.githubusercontent.com/62846458/80989070-b5eeef80-8e34-11ea-9b49-5fef90cda35c.PNG)

and on VS 2019 when I try to run some API python commands, it sends me an exception : 

![px4_21](https://user-images.githubusercontent.com/62846458/80989150-d1f29100-8e34-11ea-9181-5758ade5e96f.PNG)

Something is wrong with the GPS, but I defined them in `settings.json` as on tutorial : 

![px4_23](https://user-images.githubusercontent.com/62846458/80989401-2dbd1a00-8e35-11ea-818f-1722907b1854.PNG)

Or running PX4 in SITL rearly requires to connect and exterior device ?

**Comment by bys1123** [comment]
![image](https://user-images.githubusercontent.com/4748411/82130478-50b8d800-97fe-11ea-8e31-bfbc0a400b0e.png)
@johan606303 I failed too.

**Comment by johan606303** [comment]
@bys1123 Ok. If I find something, I will tell you.

**Comment by DanielGilo** [comment]
I'm facing similar issues, are there any updates?

**Comment by lovettchris** [comment]
Note that there are 2 PX4 messages you should be looking for:
```
INFO  [ecl/EKF] 3859668: EKF GPS checks passed (WGS-84 origin set)
INFO  [ecl/EKF] 8418081: EKF commencing GPS fusion
```
The first one does not mean it is ready to fly.  The second one means AirSIm can fly, and in AirSim you should see an additional message saying "Got a home GPS location".  AirSim cannot fly until this happens.  Sometimes this second message takes a while depending on PX4.  You can also try setting the following using the PX4 console window just in case:

```
param set LPE_LAT 47.641468
param set LPE_LON -122.140165
param set COM_OBL_ACT 1
param set NAV_RCL_ACT 0
param set NAV_DLL_ACT 0
```

just paste that block into PX4 console window and hit enter.  The AirSim settings.json file should be doing this but just in case it isn't, succeeding in passing those settings to PX4, doing it manually should also help.

**Comment by jonyMarino** [comment]
Is this related to the issues reported in #2477?

**Comment by jonyMarino** [comment]
Please try our [new version](https://github.com/microsoft/AirSim/releases) which has many updates for px4.
Closed due to age.
Please feel free to open a new issue (referring to this one) if you still have this problem.
Thanks!
